File: Week9/ProbelmA.java
/*
Language: java
ID: 63659
QLink: https://quera.org/course/assignments/63992/problems
Author: @Amirkhaskar
*/

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = Integer.parseInt(scanner.nextLine());

        for (int i = 0; i < t; i++) {
            String n = scanner.nextLine();
            String s = scanner.nextLine();

            while (s.contains("()")) {
                s = s.replace("()", "");
            }

            System.out.println(s.length() / 2);
        }
    }
}

File: Week9/ProbelmA.cs
/*
Language: csharp
ID: 63992
QLink: https://quera.org/course/assignments/63992/problems
Author: @Amirkhaskar
*/
using System;

class Program
{
    static void Main()
    {
        int t = int.Parse(Console.ReadLine());

        for (int i = 0; i < t; i++)
        {
            string n = Console.ReadLine();
            string s = Console.ReadLine();

            while (s.Contains("()"))
            {
                s = s.Replace("()", "");
            }

            Console.WriteLine(s.Length / 2);
        }
    }
}
